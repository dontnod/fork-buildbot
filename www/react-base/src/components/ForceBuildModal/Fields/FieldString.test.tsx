/*
  This file is part of Buildbot.  Buildbot is free software: you can
  redistribute it and/or modify it under the terms of the GNU General Public
  License as published by the Free Software Foundation, version 2.

  This program is distributed in the hope that it will be useful, but WITHOUT
  ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
  FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
  details.

  You should have received a copy of the GNU General Public License along with
  this program; if not, write to the Free Software Foundation, Inc., 51
  Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.

  Copyright Buildbot Team Members
*/

import renderer from 'react-test-renderer';
import { FieldString } from "./FieldString";
import { ForceSchedulerFieldString } from 'buildbot-data-js';
import { ForceBuildModalFieldsState } from '../ForceBuildModalFieldsState';

function assertRenderToSnapshot(defaultValue: string, stateValue?: string, updateValue?: string) {
  const field: ForceSchedulerFieldString = {
    name: 'dummy',
    fullName: 'fullDummy',
    label: 'dummyLabel',
    tablabel: 'dummyTabLabel',
    type: 'text',
    default: defaultValue,
    multiple: false,
    regex: null,
    hide: false,
    maxsize: null,
    autopopulate: null,
    size: 0,
  }
  const state = new ForceBuildModalFieldsState();
  state.createNewField(field.fullName, field.default);
  if (stateValue !== undefined) {
    state.setValue(field.fullName, stateValue.toString());
  }

  const component = renderer.create(
    <FieldString field={field} fieldsState={state} />
  );
  expect(component.toJSON()).toMatchSnapshot();

  if (updateValue !== undefined) {
    const expectedState = updateValue;
    renderer.act(() => {
      const elements = component.root.findAllByProps({'data-bb-test-id': `force-field-${field.fullName}`}, {deep: true});
      expect(elements.length).toBe(1);
      const input = elements[0];
      input.props.onChange({target: {value: expectedState}});
    });
    expect(state.getValue(field.fullName)).toBe(expectedState);
  }
}

describe('ForceFieldString component', function () {
  it('render default value', () => {
    assertRenderToSnapshot("");
    assertRenderToSnapshot("default");
  });

  it('render non-default value', () => {
    assertRenderToSnapshot("default", "stateValue");
  });

  it('change state on click', () => {
    assertRenderToSnapshot("default", "stateValue", "updateValue");
  });
});
