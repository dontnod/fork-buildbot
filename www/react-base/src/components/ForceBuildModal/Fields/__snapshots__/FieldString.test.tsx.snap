// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForceFieldString component change state on click 1`] = `
<div>
  <div
    className="form-group"
  >
    <div
      popover-is-open="field.haserrors"
      popover-title="{{field.label}}"
      popover-trigger="none"
      uib-popover="{{field.errors}}"
    />
    <label
      className="control-label col-sm-10"
      htmlFor="fullDummy"
    >
      dummyLabel
    </label>
    <div
      className="col-sm-10"
    >
      <input
        autoComplete="on"
        className="form-control"
        data-bb-test-id="force-field-fullDummy"
        id="fullDummy"
        onChange={[Function]}
        type="text"
        value="stateValue"
      />
    </div>
  </div>
</div>
`;

exports[`ForceFieldString component render default value 1`] = `
<div>
  <div
    className="form-group"
  >
    <div
      popover-is-open="field.haserrors"
      popover-title="{{field.label}}"
      popover-trigger="none"
      uib-popover="{{field.errors}}"
    />
    <label
      className="control-label col-sm-10"
      htmlFor="fullDummy"
    >
      dummyLabel
    </label>
    <div
      className="col-sm-10"
    >
      <input
        autoComplete="on"
        className="form-control"
        data-bb-test-id="force-field-fullDummy"
        id="fullDummy"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`ForceFieldString component render default value 2`] = `
<div>
  <div
    className="form-group"
  >
    <div
      popover-is-open="field.haserrors"
      popover-title="{{field.label}}"
      popover-trigger="none"
      uib-popover="{{field.errors}}"
    />
    <label
      className="control-label col-sm-10"
      htmlFor="fullDummy"
    >
      dummyLabel
    </label>
    <div
      className="col-sm-10"
    >
      <input
        autoComplete="on"
        className="form-control"
        data-bb-test-id="force-field-fullDummy"
        id="fullDummy"
        onChange={[Function]}
        type="text"
        value="default"
      />
    </div>
  </div>
</div>
`;

exports[`ForceFieldString component render non-default value 1`] = `
<div>
  <div
    className="form-group"
  >
    <div
      popover-is-open="field.haserrors"
      popover-title="{{field.label}}"
      popover-trigger="none"
      uib-popover="{{field.errors}}"
    />
    <label
      className="control-label col-sm-10"
      htmlFor="fullDummy"
    >
      dummyLabel
    </label>
    <div
      className="col-sm-10"
    >
      <input
        autoComplete="on"
        className="form-control"
        data-bb-test-id="force-field-fullDummy"
        id="fullDummy"
        onChange={[Function]}
        type="text"
        value="stateValue"
      />
    </div>
  </div>
</div>
`;
